src/core/ipns/routing/pubsub.js

```diff
@@ -0,0 +1,206 @@
+'use strict'
+
+const ipns = require('ipns')
+const { fromB58String, toB58String } = require('multihashes')
+const PeerId = require('peer-id')
+const DatastorePubsub = require('datastore-pubsub')
+
+const errcode = require('err-code')
+const debug = require('debug')
+const log = debug('jsipfs:ipns:pubsub')
+log.error = debug('jsipfs:ipns:pubsub:error')
+
+const ipnsNS = '/ipns/'
+const ipnsNSLength = ipnsNS.length
+
+// Pubsub aims to manage the pubsub subscriptions for IPNS
+class Pubsub {
+  constructor (node) {
+    const pubsub = node._libp2pNode.pubsub
+    const localDatastore = node._repo.datastore
+    const peerId = node._peerInfo.id
```

Comment:
I'd pass these directly to the constructor so that it is more explicit what dependencies this class has.